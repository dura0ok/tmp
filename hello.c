#include "postgres.h"
#include "fmgr.h"
#include "tcop/utility.h"

#include "utils/builtins.h"

char *message = "hello, c world!";
const char* need = "create table";
static ProcessUtility_hook_type prev_ProcessUtility = NULL;
static int counter;

PG_MODULE_MAGIC;

Datum hello_cworld(PG_FUNCTION_ARGS);

PG_FUNCTION_INFO_V1(hello_cworld);

Datum
hello_cworld(PG_FUNCTION_ARGS)
{
    PG_RETURN_INT32(counter);
}

static void pgss_ProcessUtility(PlannedStmt *pstmt, const char *queryString,
                                bool readOnlyTree,
                                ProcessUtilityContext context, ParamListInfo params,
                                QueryEnvironment *queryEnv,
                                DestReceiver *dest, QueryCompletion *qc);

void
_PG_init()
{   
    counter = 0;
    prev_ProcessUtility = ProcessUtility_hook;
    ProcessUtility_hook = pgss_ProcessUtility;
}


static void
pgss_ProcessUtility(PlannedStmt *pstmt, const char *queryString,
                    bool readOnlyTree,
                    ProcessUtilityContext context,
                    ParamListInfo params, QueryEnvironment *queryEnv,
                    DestReceiver *dest, QueryCompletion *qc)
{
   if (prev_ProcessUtility != NULL){
                prev_ProcessUtility(pstmt, queryString, readOnlyTree,
                                    context, params, queryEnv,
                                    dest, qc);
    }else{
        standard_ProcessUtility(pstmt, queryString, readOnlyTree,
                                        context, params, queryEnv,
                                        dest, qc);
    }
   
   elog(LOG, "Hi");
   elog(LOG, queryString);
   int result = strncasecmp(queryString, need, strlen(need));
   if(result == 0){
      elog(LOG, "asdasd ===========================");
      counter++;
   }
}