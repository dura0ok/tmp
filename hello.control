# fuzzystrmatch extension
comment = 'test extension'
default_version = '1.0'
module_pathname = '$libdir/hello'
relocatable = true
