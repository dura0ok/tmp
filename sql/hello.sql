create extension hello;
select hello_world();
select hello_cworld();
create table kek(t int);
select hello_cworld();
create table kek(t int);
select hello_cworld();
drop table kek;
select hello_cworld();