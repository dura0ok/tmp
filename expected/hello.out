  hello_world  
---------------
 Hello, world!
(1 row)
 hello_cworld 
--------------
            0
(1 row)
 hello_cworld 
--------------
            1
(1 row)
 hello_cworld 
--------------
            1
(1 row)
 hello_cworld 
--------------
            1
(1 row)
